transformer_engine/pytorch/utils.py

```diff
@@ -305,3 +305,12 @@ def devices_match(device1: torch.device, device2: torch.device) -> bool:
             index2 = torch.cuda.current_device()
         return index1 == index2
     return device1 == device2
+
+
+def torch_get_autocast_gpu_dtype() -> torch.dtype:
+    """Get PyTorch autocast GPU dtype."""
+    TORCH_MAJOR = int(torch.__version__.split(".")[0])
+    TORCH_MINOR = int(torch.__version__.split(".")[1])
+    if TORCH_MAJOR == 2 and TORCH_MINOR >= 4:
```

Comment:
Could we do a simpler >= "2.4" similar to the one currently in jit.py. This method isn't future proof and will break for 3+

Reply:
I'm good with it.
Furthermore, since we now have `packaging` as a dependency, how about using `packaging.version` for a more robust version comparison?